nixos/modules/tasks/filesystems/zfs.nix

```diff
@@ -623,6 +629,9 @@ in
               fi
               poolImported "${pool}" || poolImport "${pool}"  # Try one last time, e.g. to import a degraded pool.
             fi
+
+            ${concatMapStringsSep "\n" (elem: "clevis decrypt < /etc/clevis/${elem}.jwe | zfs load-key ${elem}") (filter (p: (elemAt (splitString "/" p) 0) == pool) clevisDatasets)}
```

Comment:
Wouldn't this add calls to `clevis` even when `boot.initrd.clevis.enable` is false?

I think that's why I started getting this evaluation error yesterday: https://gist.github.com/aij/bb5c49f0afbde2b3efdfc8f30c1e7b6e

Reply:
Thank you for reporting that, I am proposing a fix here: https://github.com/NixOS/nixpkgs/pull/272061